### DISCUSSION TITLE
Same name opa tests

### DISCUSSION THREAD

**Discussion by wermerb** [open-policy-agent/community#331] [OPA and Rego] [answered]
Hi,

Lets say i have the following test rego:
```
package app

test_allow {
    true
}

test_allow {
    false
}
```
When i run `opa test test.rego` I saw the following output:
```
$ opa test test.rego
$ test.rego:
$ data.app.test_allow#01: FAIL (0s)
$ --------------------------------------------------------------------------------
$ PASS: 1/2
$ FAIL: 1/2
```
So this gave me this the idea, what if i just want to re-test the one which is failed, so i ran the following:
`opa test -r test_allow#01 test.rego`
```
$ opa test -r "test_allow#01" test.rego
$ test.rego:
$ data.app.test_allow#01: FAIL (0s)
--------------------------------------------------------------------------------
FAIL: 1/1
```
Out of curiosity i wanted to try to run the first test case with the following command: `opa test -r test_allow#00 test.rego`
but i got no output at all. Is there any way to reference the first test in my case somehow?

**Comment by charlieegan3** [comment, marked as answer]
Hey, this is a bit of an edge case but I think you can run `opa test -r "data.app.test_allow" test.rego -v` to only run the first one. However, I'd recommend that the test cases are named if you need to run specific tests, e.g.

```rego
package app

test_allow_with_ok_input {
    true
}

test_allow_with_bad_input {
    false
}
```
